Error [INSTALL_PARSE_FAILED_INCONSISTENT_CERTIFICATES] not shown with multiple devices

I noticed (actually spend some time trying to figure out what was going on) that when you have more than one device connected and the app you want to test is install with a different certificate the error message:

    app apk install failed.  Error [INSTALL_PARSE_FAILED_INCONSISTENT_CERTIFICATES]

is not shown, instead you will just see this:

    2014-02-04 11:11:09 [SR.runTests] Executing instrumentation suite on 2 device(s).
    
    * What went wrong:
    Execution failed for task ***.
    > Tests failed! See spoon/debug/index.html

If there is only one device connected you will see this:

    2014-02-04 10:58:51 [SR.runTests] Executing instrumentation suite on 1 device(s).
    2014-02-04 10:58:57 [SDR.run] [02e7ea272138b62a] app apk install failed.  Error [INSTALL_PARSE_FAILED_INCONSISTENT_CERTIFICATES]
    :*** FAILED

    FAILURE: Build failed with an exception.

    * What went wrong:
    Execution failed for task ***.
    > Tests failed! See spoon/debug/index.html

It seems with two devices it silently crashes and generates the web page, which then has no tests on it, even the device with the same certificate won't have the tests executed.

I'll try to understand whats going on in the code, any thoughts about it are appreciated!

--- a/spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java
+++ b/spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java
@@ -167,6 +167,7 @@ public final class SpoonRunner {
             try {
               summary.addResult(safeSerial, getTestRunner(serial, testInfo).runInNewProcess());
             } catch (Exception e) {
+              e.printStackTrace(System.out);
               summary.addResult(safeSerial, new DeviceResult.Builder().addException(e).build());
             } finally {
               done.countDown();

